api/client/lib.py

```diff
@@ -232,7 +244,10 @@ def get_data(url, headers, params=None, logger=None):
             logger.warning('Redirecting {} to {}'.format(params, new_params), extra=log_record)
             params = new_params
         elif data.status_code in [400, 401, 404, 500]:
-            break
+            try:
+                raise APIError(data, retry_count, url)
+            except APIError as error:
+                return error
         else:
             logger.error('{}'.format(data), extra=log_record)
     raise Exception('Giving up on {} after {} tries: {}.'.format(
```

Comment:
Let's also raise an APIError here. This one's if you're timing out repeatedly and it gives up, so `data` in this case would likely be a 503 or 504 error.

Reply:
Done!